src/redux/wallet/wallet.js

```diff
@@ -75,7 +85,38 @@ const transferLht = (amount, recipient) => (dispatch) => {
 const transferTime = (amount, recipient) => (dispatch) => {
     dispatch(setTimeBalanceStart());
     TimeProxyDAO.transfer(amount, recipient, localStorage.getItem('chronoBankAccount'))
-        .then(() => dispatch(updateTimeBalance()));
+        .then(() => dispatch(updateTimeBalance(account)));
+};
+
+const requireTime = (account) => (dispatch) => {
+    return TokenContractsDAO.requireTime(account).then((r) => {
+        if(r){
+            dispatch(showAlertModal({title: 'Require Time', message: 'Time request sent successfully.'}));
+            return dispatch(updateTimeBalance(account))
+        } else {
+            dispatch(showAlertModal({title: 'Error', message: 'Time request not completed.'}));
+        }
+    });
+};
+
+const depositTime = (amount, account, successAction = () => {}) => (dispatch) => {
+    return TimeHolderDAO.depositAmount(amount, account).then((r) => {
+        if (r) {
+            successAction();
```

Comment:
Your action creator should be pristine, so using such callback here is prohibited. You don't need to pass such callbacks here, you can dispatch corresponding action creator right here, which will hide your modal or do something else.